src/libraries/System.Formats.Tar/tests/TarFile/TarFile.ExtractToDirectoryAsync.File.Tests.cs

```diff
@@ -56,6 +56,33 @@ public async Task NonExistentDirectory_Throws_Async()
             }
         }
 
+        [Fact]
+        public async Task SetsLastModifiedTimeOnExtractedFiles()
+        {
+            using TempDirectory root = new TempDirectory();
+
+            string inDir = Path.Join(root.Path, "indir");
+            string inFile = Path.Join(inDir, "file");
+
+            string tarFile = Path.Join(root.Path, "file.tar");
+
+            string outDir = Path.Join(root.Path, "outdir");
+            string outFile = Path.Join(outDir, "file");
+
+            Directory.CreateDirectory(inDir);
+            File.Create(inFile).Dispose();
+            var dt = new DateTime(2001, 1, 2, 3, 4, 5, DateTimeKind.Local);
+            File.SetLastWriteTime(inFile, dt);
+
+            await TarFile.CreateFromDirectoryAsync(sourceDirectoryName: inDir, destinationFileName: tarFile, includeBaseDirectory: false);
+
+            Directory.CreateDirectory(outDir);
+            await TarFile.ExtractToDirectoryAsync(sourceFileName: tarFile, destinationDirectoryName: outDir, overwriteFiles: false);
+
+            Assert.True(File.Exists(outFile));
+            Assert.InRange(File.GetLastWriteTime(outFile).Ticks, dt.AddSeconds(-3).Ticks, dt.AddSeconds(3).Ticks); // include some slop for filesystem granularity
```

Comment:
Hope this range is enough. We'll see what the CI says.

Reply:
I copied this from some IO tests. The worst is FAT, which has a 3 second granularity. So this should be more than enough.